src/main/scala/nl/biopet/bioconda/BiocondaPlugin.scala

```diff
@@ -162,51 +190,98 @@ object BiocondaPlugin extends AutoPlugin {
   }
 
   /**
-    * A task that returns a sequence of net yet published tags and the latest tag
-    * @return sequence of not yet published tags, latest tag
+    * Gets the released tags from github and determines the latest
+    * @return the latest to be released tag.
     */
-  private def getUnPublishedTags: Def.Initialize[Task[R]] = {
+  private def getLatestTag: Def.Initialize[Task[TagName]] = {
     Def.task {
-      val publishedTags: Seq[TagName] = getPublishedTags.value
-      val releasedTags: Seq[TagName] = getReleasedTags.value
-      val latestTag = releasedTags
-        .sortBy(tag => tag.stripPrefix("v"))
+      getReleasedTags.value
+        .sortBy(tag =>
+          SemanticVersion.fromString(tag) match {
+            case Some(tagVersion) => tagVersion
+            case _                => new SemanticVersion(0, 0, 0)
+        })
         .lastOption
         .getOrElse("No version")
-      // Tags that are released but not in bioconda yet should be published
-      // make sure latest tag is always published if latest.
-      val unPublishedTags: Seq[TagName] =
-        releasedTags.filter(tag => !publishedTags.contains(tag)).distinct
-      R(unPublishedTags, latestTag)
     }
   }
-  private case class R(unPublished: Seq[TagName], latest: TagName)
 
-  private def createRecipes(
-      latest: Boolean = true,
-      versions: Boolean = true): Def.Initialize[Task[File]] = {
+  /**
+    * Create the recipe for the current version
+    * @return the directory with the recipes
+    */
+  private def createCurrentRecipe(): Def.Initialize[Task[File]] = {
+    Def.taskDyn {
+      val tag = "v" + version.value
+      val releasedTags = getReleasedTags.value
+      if (!releasedTags.contains(tag)) {
+        throw new Exception(
```

Comment:
A general exception like this not best practice, it's better to raise a A IllegalStateException here